lib/realtime_web/channels/realtime_channel.ex

```diff
@@ -586,4 +604,38 @@ defmodule RealtimeWeb.RealtimeChannel do
     Logger.error("Start channel error: " <> error_msg)
     {:error, %{reason: error_msg}}
   end
+
+  defp assign_policies(false, db_conn, sub_topic, access_token, claims, socket) do
+    case ChannelsCache.get_channel_by_name(sub_topic, db_conn) do
+      {:error, :not_found} ->
+        {:error, "Channel #{sub_topic} does not exist, please create it first"}
+
+      {:ok, channel} ->
+        authorization_context =
+          Authorization.build_authorization_params(%{
+            channel: channel,
+            headers: socket.assigns.headers,
+            jwt: access_token,
+            claims: claims,
+            role: claims["role"]
+          })
+
+        {:ok, socket} = Authorization.get_authorizations(socket, db_conn, authorization_context)
+
+        case socket.assigns.policies do
+          %Policies{broadcast: %BroadcastPolicies{read: false}} ->
+            {:error, "You do not have permissions to access this broadcast"}
+
+          %Policies{channel: %ChannelPolicies{read: false}} ->
+            {:error, "You do not have permissions to access this channel"}
```

Comment:
When would you have access to a Channel and not have access to Broadcast? 

Probably never I guess but you -could- write an RLS policy this way.

Reply:
Maybe "You do not have permissions to read Broadcast messages on this Channel"

And "You do not have permissions to read this Channel"

Need to be very specific here in our error responses.

Technically you could have permission to write Broadcast messages to a Channel but not read them.

Reply:
> When would you have access to a Channel and not have access to Broadcast?
> Probably never I guess but you -could- write an RLS policy this way.
this two things are "decoupled" as we pass the channel in the context and the channel is fetched as a superuser. We could link both 👀 

> Need to be very specific here in our error responses.

Yeah makes sense to go with more specific error messages as this will be confusing